syntax = "proto3";

package buildFirmwarePackage;

service BuildFirmwareServer {
  rpc buildFirmware(BuildFirmwareRequest) returns (stream BuildFirmwareResponse) {}
}

message BuildFirmwareRequest {
    string uuid = 1;
    string projectName = 2;
    string hlsToolkit  = 3;
    string targetBoard = 4;
    string arch = 5;
    string projectType = 6;
    int32  nInputs = 7;
    int32  nOutputs = 8;
    string flavor = 9;
    string sourceNeuralNetwork = 10;
}
  
message BuildFirmwareResponse {
    bool success = 1;
    string message = 2;
}
  